spec/lib/vanagon/component_spec.rb

```diff
@@ -45,20 +46,32 @@
       @workdir = Dir.mktmpdir
       @file_name = 'fake_file.txt'
       @fake_file = "file://spec/fixtures/files/#{@file_name}"
+      @fake_dir = 'fake_dir'
+      @fake_tar = "file://spec/fixtures/files/#{@fake_dir}.tar.gz"
     end
 
     subject do
       # Initialize a new instance of Vanagon::Component and define a
       # new secondary source. We can now reason about this instance and
       # test behavior for retrieving secondary sources.
-      Vanagon::Component.new('build-dir-test', {}, {}).tap do |comp|
-        comp.sources << OpenStruct.new(url: @fake_file)
-      end
+
+      plat = Vanagon::Platform::DSL.new('el-5-x86_64')
+      plat.instance_eval("platform 'el-5-x86_64' do |plat| end")
+      @platform = plat._platform
+
+      comp = Vanagon::Component::DSL.new('build-dir-test', {}, @platform)
+      comp.add_source @fake_file
+      comp.add_source @fake_tar
+      comp._component
     end
 
     it "copies secondary sources into the workdir" do
-      subject.get_sources(@workdir)
+      component = subject
+      component.get_sources(@workdir)
       expect(File.exist?(File.join(@workdir, @file_name))).to be true
+      # make sure that our secondary source(s) made it into the workdir
+      expect(File.exist?(File.join(@workdir, "#{@fake_dir}.tar.gz"))).to be true
+      expect(component.extract_with.join(" && ")).to match "#{@fake_dir}.tar.gz"
     end
```

Comment:
Instead of modifying the existing test, considering adding a new test that validates that the compressed files are expanded. Leave the existing test for uncompressed files alone, because the expectation is that secondary sources can be either kind of file.